src/prefect/server/api/server.py

```diff
@@ -671,14 +673,9 @@ async def start_services():
     async def stop_services():
         """Ensure services are stopped before the Prefect REST API shuts down."""
         if hasattr(app.state, "services") and app.state.services:
-            await asyncio.gather(*[service.stop() for service in app.state.services])
-            try:
-                await asyncio.gather(
-                    *[task.stop() for task in app.state.services.values()]
-                )
-            except Exception:
-                # `on_service_exit` should handle logging exceptions on exit
-                pass
+            service_tasks: dict[Service, asyncio.Task[None]] = app.state.services
+            await asyncio.gather(*[service.stop() for service in service_tasks])
+            await asyncio.gather(*service_tasks.values(), return_exceptions=True)
```

Comment:
I believe our prior implementation with `task.stop()` was just quietly failing to await the tasks.  Here with `return_exceptions` we can await them all and ignore their errors.